### PR TITLE
Add klazz helper to Controls Dsl

### PR DESCRIPTION
**Pull request by jonspalmer**
I think this extra helper method is useful for the common case of building a class from control driven params.  Curious as to others views.

### CODE REVIEW

File: spec/view_component/storybook/dsl/controls_dsl_spec.rb [hunk 1]
```diff
@@ -142,4 +142,43 @@
...
+
+    it "returns cutom value from params" do
+      from_params = subject.value_from_params({})
+      expect(from_params).to eq("J.R.R. Tolkien")
+    end
+  end
+
+  describe "#klazz" do
+    subject do
+      klazz(Author, first_name: "J.R.R.", last_name: "Tolkien").param(:author)
```
Comment (hjhart) [inline spec/view_component/storybook/dsl/controls_dsl_spec.rb:168]: Seeing the `.param(author)` so late in the line seemed a bit off to me... potentially a rearrangement might look like `klazz(:author, Author, ...`.

I got a bit tripped up by the syntax the first time reading it. Or maybe we just call it `custom_param(:author, Author, ...`?

File: spec/view_component/storybook/dsl/controls_dsl_spec.rb [hunk 2]
```diff
@@ -142,4 +142,43 @@
...
+                       class: ViewComponent::Storybook::Controls::CustomConfig,
+                       param: :author,
+                       name: "Author"
+                     }
+
+    it "returns cutom value from params" do
+      from_params = subject.value_from_params({})
+      expect(from_params).to eq("J.R.R. Tolkien")
+    end
+  end
```
Comment (hjhart) [inline spec/view_component/storybook/dsl/controls_dsl_spec.rb:164]: Great to add this spec for sure.

File: spec/view_component/storybook/dsl/controls_dsl_spec.rb (hunk 1 above)
Comment (hjhart) [reply spec/view_component/storybook/dsl/controls_dsl_spec.rb:168]: Ignore me. I'm realizing that this DSL is for the spec. I looked back at #50 again to better understand, and I like the klazz shorthand now. :)

Comment (jonspalmer) [reply spec/view_component/storybook/dsl/controls_dsl_spec.rb:168]: Yeah. Honestly there is very little reason I can think of that a consumer would actually need to call `param` themselves. We infer it from the name of args anyway. Maybe if you're using `*args` or `**kwargs` then you'd want to set it? But even then that actual over the wire params is mostly an internal detail. It's more likely you want to override the name.

For these specs the only reason I'm including the param is so we can a param and a name to test against. Even then maybe I should drop it from these specs as its confusing?

File: spec/view_component/storybook/dsl/controls_dsl_spec.rb (hunk 2 above)
Comment (jonspalmer) [reply spec/view_component/storybook/dsl/controls_dsl_spec.rb:164]: missed it the first time around 😞

**Pull request merged**